crates/steel-core/src/rvals/cycles.rs

```diff
@@ -216,54 +174,13 @@ impl CycleDetector {
                     }
                 }
             }
-
-            PortV(_) => write!(f, "#<port>"),
-            Closure(_) => write!(f, "#<bytecode-closure>"),
-            HashMapV(hm) => write!(f, "#<hashmap {:#?}>", hm.as_ref()),
-            IterV(_) => write!(f, "#<iterator>"),
-            HashSetV(hs) => write!(f, "#<hashset {:?}>", hs.0),
-            FutureFunc(_) => write!(f, "#<future-func>"),
-            FutureV(_) => write!(f, "#<future>"),
-            StreamV(_) => write!(f, "#<stream>"),
-            BoxedFunction(b) => {
-                if let Some(name) = b.name() {
-                    write!(f, "#<function:{}>", name)
-                } else {
-                    write!(f, "#<function>")
-                }
-            }
-            ContinuationFunction(_) => write!(f, "#<continuation>"),
-            // #[cfg(feature = "jit")]
-            // CompiledFunction(_) => write!(f, "#<compiled-function>"),
-            ListV(l) => {
-                write!(f, "(")?;
-
-                let mut iter = l.iter().peekable();
-
-                while let Some(item) = iter.next() {
-                    self.format_with_cycles(item, f)?;
-                    if iter.peek().is_some() {
-                        write!(f, " ")?
-                    }
-                }
-
-                write!(f, ")")
-            }
-            MutFunc(_) => write!(f, "#<function>"),
-            BuiltIn(_) => write!(f, "#<function>"),
-            ReducerV(_) => write!(f, "#<reducer>"),
-            MutableVector(v) => write!(f, "{:?}", v.get()),
-            SyntaxObject(s) => {
-                if let Some(raw) = &s.raw {
-                    write!(f, "#<syntax:{:?} {:?}>", s.span, raw)
-                } else {
-                    write!(f, "#<syntax:{:?} {:?}>", s.span, s.syntax)
-                }
-            }
-            BoxedIterator(_) => write!(f, "#<iterator>"),
-            Boxed(b) => write!(f, "'#&{}", b.borrow()),
-            Reference(x) => write!(f, "{}", x.format()?),
             HeapAllocated(b) => write!(f, "'#&{}", b.get()),
+            _ => {
+                self.depth -= 1;
+                let res = self.format_with_cycles(val, f);
+                self.depth += 1;
+                res
+            }
```

Comment:
I'm having some trouble following this change - what was the purpose here?

Reply:
Redid this part, I think this is cleaner. I can also revert back to having to separate but similar functions.

- High level: format_with_cycles and top_level_format_with cycles seem to be very similar ([difference](https://www.diffchecker.com/3X0rhqAZ/)) so tried to unify.
- Previously reviewed approach: Reused one of the functions when the special cases didn't match. But this means the depth was artificially increased a level. Probably doesn't matter but tried to keep the depth anyways.
- New approach - Pass an enum specifying which behavior to use.

Reply:
got it, I understand, I haven't looked at some of these functions in a long time :smile: